.github/workflows/deploy.yml

```
@@ -0,0 +1,53 @@
+          cache: 'npm'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build for preview
+        run: npm run generate
+
+      - name: Comment PR with preview info
+        uses: actions/github-script@v7
+        with:
+          script: |
```

actionlint (1.7.11)

[error] "github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)